mri/utils/process_SWI

```diff
@@ -0,0 +1,123 @@
+# -*- coding: utf-8 -*-
+##########################################################################
+# pySAP - Copyright (C) CEA, 2017 - 2018
+# Distributed under the terms of the CeCILL-B license, as published by
+# the CEA-CNRS-INRIA. Refer to the LICENSE file or to
+# http://www.cecill.info/licences/Licence_CeCILL-B_V1-en.html
+# for details.
+##########################################################################
+
+import numpy as np
+
+
+def virtual_coil_reconstruction(imgs):
+    """
+    Calculate the combination of all coils using virtual coil reconstruction
+
+    Parameters
+    ----------
+    imgs: np.ndarray
+        The images reconstructed channel by channel
+        in shape [Nch, Nx, Ny, Nz] for 3D or [Nch, Nx, Ny] for 2D
+
+    Returns
+    -------
+    img_comb: np.ndarray
+        The combination of all the channels in a complex valued
+        in shape [Nx, Ny, Nz] for 3D or [Nx, Ny] for 2D
+    """
+    img_sh = imgs.shape
+    # Compute first the virtual coi
+    weights = np.sum(np.abs(imgs), axis=0)
+    weights[weights == 0] = 1e-16
+    phase_reference = np.asarray([np.angle(np.sum(
+        imgs[ch].flatten())) for ch in range(img_sh[0])])
+    reference = np.asarray([(imgs[ch] / weights) / np.exp(
+        1j * phase_reference[ch]) for ch in range(32)])
+    virtual_coil = np.sum(reference, axis=0)
+    difference_original_vs_virtual = np.conjugate(imgs) * virtual_coil
+    # Hanning filtering in readout and phase direction
+    hanning = get_hanning_filter([img_sh[1], img_sh[2]])
+    if len(img_sh) > 3:
+        hanning = np.tile(np.expand_dims(np.tile(
```

Comment:
Cant we add this inside get_hanning_filter? 
Maybe we just send image shape and get a complete hanning filter?

Reply:
There is no hanning filtering in the partition direction (z), plus the hanning filter is only applied on a portion of the 2D kspace before to be extended in the third dimension.
If we want to keep the get_hanning_filter general, this part can't be include in the function because it is really specific to the SWI process.